components/script/dom/htmlimageelement.rs

```diff
@@ -267,33 +267,31 @@ impl HTMLImageElement {
     fn fetch_image(&self, img_url: &ServoUrl) {
         let window = window_from_node(self);
         let image_cache = window.image_cache();
-        let response = image_cache.find_image_or_metadata(
-            img_url.clone().into(),
-            UsePlaceholder::Yes,
-            CanRequestImages::Yes,
-        );
-        match response {
-            Ok(ImageOrMetadataAvailable::ImageAvailable(image, url)) => {
-                self.process_image_response(ImageResponse::Loaded(image, url));
-            },
 
-            Ok(ImageOrMetadataAvailable::MetadataAvailable(m)) => {
-                self.process_image_response(ImageResponse::MetadataLoaded(m));
+        match image_cache.get_image(&img_url, UsePlaceholder::Yes) {
+            Some(img) => {
+                return self.process_image_response(ImageResponse::Loaded(img, img_url.clone()))
             },
+            None => (),
+        }
 
-            Err(ImageState::Pending(id)) => {
-                add_cache_listener_for_element(image_cache, id, self);
-            },
+        let cache_result =
+            image_cache.track_image(img_url.clone(), UsePlaceholder::Yes, CanRequestImages::Yes);
```

Comment:
We should be able to merge this with the earlier get_image call based on my previous comments.